### DISCUSSION TITLE
Overlay API uploadOverlayImage fails if multipart boundary is over 43 characters

### DISCUSSION THREAD

**Discussion by Manezki** [AxisCommunications/discussions#242] [VAPIX] [answered]
Hey. My team and I are wrapping some of the Vapix endpoints into an internal Go library. While doing so we found out that the Overlay API fails if the multipart boundary is over 43 characters long. I'm reporting this, as we could not find documentation related to multipart request limits, nor did the Vapix responses help us diagnose the problem.

## Vapix responses with varying boundary length

Below calls are made to [Overlay API uploadOverlayImage endpoint](https://www.axis.com/vapix-library/subjects/t10175981/section/t10007638/display?section=t10007638-t10138362).

When the Multipart boundary is 44 characters (Go's standard library's Multipart has a boundary length of 60):
```
{
  "apiVersion": "1.0",
  "method": "unknown",
  "error": {
    "code": 1004,
    "message": "invalid request"
  }
}
```

When the Multipart boundary is 43 characters:
```
{
  "apiVersion": "1.0",
  "method": "uploadOverlayImage",
  "data": {
    "path": "/etc/overlays/testpic(228x140).ovl"
  }
}
```

What makes the situation difficult is that the response error 1004 is documented on [Overlay API](https://www.axis.com/vapix-library/subjects/t10175981/section/t10007638/display?section=t10007638-t10140695) with description 'The provided input was invalid.'.

## Tested on

Camera type: P3265-V
Firmware: 11.4.63

**Comment by vivekatoffice** [comment]
Hi @Manezki ,
Have you had a chance to explore potential solutions or make any progress? Your insights would be greatly appreciated.

**Reply by Manezki** [reply, marked as answer]
Hey @vivekatoffice

As a workaround we limit the boundary size to 40 characters on the client Go code:
``` go
body := &bytes.Buffer{}
writer := multipart.NewWriter(body)
writer.SetBoundary(writer.Boundary()[0:40])
```

The workaround hasn't caused any problems that I'm aware of.

I haven't found Vapix-related repositories on the public Axis Communications repositories, so I haven't looked into serverside-fix. Do you @vivekatoffice happen to know if Vapix is included in one of the repositories?

**Reply by vivekatoffice** [reply]
Hi @Manezki ,
No, these repositories are not public. I will pass your feedback to the team to mention the correct character limits.
